src/builtins/core/date.rs

```diff
@@ -651,6 +651,45 @@ impl PlainDate {
             .with_calendar(self.calendar.identifier(), display_calendar)
             .build()
     }
+
+    #[inline]
+    pub fn to_zoned_date_time_with_provider(
+        &self, 
+        tz: TimeZone,
+        plain_time: Option<PlainTime>, 
+        provider: &impl TimeZoneProvider
+    ) -> TemporalResult<ZonedDateTime> {
+        // 1. Let temporalDate be the this value.
+        // 2. Perform ? RequireInternalSlot(temporalDate, [[InitializedTemporalDate]]).
+        //3. If item is an Object, then
+            //a. Let timeZoneLike be ? Get(item, "timeZone").
+            //b. If timeZoneLike is undefined, then
+            //   i. Let timeZone be ? ToTemporalTimeZoneIdentifier(item).
+            //    ii. Let temporalTime be undefined.
+            // c. Else,
+            //    i. Let timeZone be ? ToTemporalTimeZoneIdentifier(timeZoneLike).
+            //    ii. Let temporalTime be ? Get(item, "plainTime").
+        // 4. Else,
+        //     a. Let timeZone be ? ToTemporalTimeZoneIdentifier(item).
+        //     b. Let temporalTime be undefined.
+
+        //  5. If temporalTime is undefined, then
+        //     a. Let epochNs be ? GetStartOfDay(timeZone, temporalDate.[[ISODate]]).
+        //  6. Else,
+        //     a. Set temporalTime to ? ToTemporalTime(temporalTime).
+        //     b. Let isoDateTime be CombineISODateAndTimeRecord(temporalDate.[[ISODate]], temporalTime.[[Time]]).
+        //     c. If ISODateTimeWithinLimits(isoDateTime) is false, throw a RangeError exception.        
+        //     d. Let epochNs be ? GetEpochNanosecondsFor(timeZone, isoDateTime, compatible).
+        let epoch_ns = if let Some(time) = plain_time {    
+            let temporal_time = Time::new_unchecked(time)?;
```

Comment:
issue: I'm not entirely sure what the time is referring to here. But it should be unneeded.

You should be able to access `PlainTime`'s iso field by using `time.iso` in line 685.